library/core/src/slice/mod.rs

```diff
@@ -865,8 +865,10 @@ impl<T> [T] {
     pub fn chunks_exact(&self, chunk_size: usize) -> ChunksExact<'_, T> {
         assert_ne!(chunk_size, 0);
         let rem = self.len() % chunk_size;
-        let len = self.len() - rem;
-        let (fst, snd) = self.split_at(len);
+        let fst_len = self.len() - rem;
+        // The following is the same as self.split_at(fst_len) but avoids a bounds check
+        // SAFETY: 0 <= fst_len < self.len() by construction above
+        let (fst, snd) = unsafe { (self.get_unchecked(0..fst_len), self.get_unchecked(fst_len..)) };
```

Comment:
Drive by comment: maybe `self.get_unchecked(..fst_len)` here and elsewhere?

Reply:
Wait, I believe this is not `scd`, it should be `rem` / `rest`. `scd` is second but it does not means second here, it means all items other than first.